Flaky cleanup race in cmd_stop — lock file not reliably removed after SIGKILL, blocks every PR's CI

## What happened

\`make gate\` on main is non-deterministically failing in two related tests:

\`\`\`
FAILED tests/test_cli_commands.py::TestCmdStop::test_timeout_escalates_to_sigkill_and_cleans_up
FAILED tests/test_cli_commands.py::TestCmdStop::test_timeout_escalates_to_sigkill_and_cleans_up_sprint_locks
\`\`\`

Failure detail:

\`\`\`
>       assert not lock_path.exists()
E       AssertionError: assert not True
E        +  where True = exists()
E        +    where exists = PosixPath('.../.forge/locks/my-slug.lock').exists
\`\`\`

The test calls \`cmd_stop\`, which is supposed to escalate SIGTERM → SIGKILL on a stuck run and clean up its lock file. The assertion checks that after \`cmd_stop\` returns, the lock file no longer exists. Sometimes it does, sometimes it doesn't. Same commit, no test changes — it's a race.

Concrete observations:
- Earlier in a session: \`make gate\` from current main → 3641 passed
- Same commit, 30 minutes later: \`make gate\` → 3639 passed, 2 failed (the two above)
- CI on every PR (e.g. #1111) hits the same failures consistently — the merge queue is effectively blocked
- This is also a plausible cause of \"Gate timed out after 45s\" escalations seen on #1070's sprint runs: if the cleanup path can race into a blocked-not-failed state under some conditions, the test hangs instead of failing

## What was expected

After \`cmd_stop\` returns 0, the lock file at \`<runs_dir>/.forge/locks/<slug>.lock\` is reliably absent. No race window where the lock can persist after a successful stop.

The two test failures should be deterministic across runs. \`make gate\` on a clean main checkout passes 100% of the time, not 50% or 90%.

## Notes

- The test setup mocks \`os.kill\` and \`_is_pid_alive\` so the SIGTERM/SIGKILL signaling is synchronous in test. The race is therefore inside \`cmd_stop\`'s cleanup logic, not in actual process signaling — likely a thread/async cleanup path that the function returns before completing.
- This blocks v0.9.0 directly: every PR's CI fails on this, the auto-merge pipeline is jammed, and sprint runs see opaque gate timeouts that look like environment problems but are really this race propagating into hung tests.
- Related but distinct: #1115 (gate hardening with per-test timeouts) would convert the worst-case hang into a clean per-test failure, but does not address the underlying race.

--- a/src/theforge/sprint/lock.py
+++ b/src/theforge/sprint/lock.py
@@ -9,8 +9,8 @@ import time
 from contextlib import contextmanager
 from pathlib import Path
 
+from theforge import detach as _detach_mod
 from theforge.artifacts import ESCALATED_MARKER_PATH
-from theforge.detach import _is_pid_alive
 from theforge.pid import _current_process_fingerprint, _pid_matches_fingerprint
 
 _LOCK_METADATA_SEPARATOR = "|"
@@ -233,9 +233,9 @@ def cleanup_story_locks(
                     if owner_fingerprint is None:
                         should_remove = True
                     else:
-                        should_remove = not _is_pid_alive(owner_pid) or _pid_matches_fingerprint(
-                            owner_pid, owner_fingerprint
-                        )
+                        should_remove = not _detach_mod._is_pid_alive(
+                            owner_pid
+                        ) or _pid_matches_fingerprint(owner_pid, owner_fingerprint)
             if should_remove:
                 lock_path.unlink(missing_ok=True)
                 cleaned.append(slug)

--- a/tests/test_sprint_lock.py
+++ b/tests/test_sprint_lock.py
@@ -264,7 +264,7 @@ class TestCleanupStoryLocks:
         lock_path.write_text("424242|old-start\n", encoding="utf-8")
 
         with (
-            patch("theforge.sprint.lock._is_pid_alive", return_value=False),
+            patch("theforge.detach._is_pid_alive", return_value=False),
             patch("theforge.sprint.lock._pid_matches_fingerprint", return_value=False),
         ):
             cleaned = cleanup_story_locks(["story-a"], tmp_path, pid=424242)
@@ -278,7 +278,7 @@ class TestCleanupStoryLocks:
         lock_path.write_text("424242|other-start\n", encoding="utf-8")
 
         with (
-            patch("theforge.sprint.lock._is_pid_alive", return_value=True),
+            patch("theforge.detach._is_pid_alive", return_value=True),
             patch("theforge.sprint.lock._pid_matches_fingerprint", return_value=False),
         ):
             cleaned = cleanup_story_locks(["story-a"], tmp_path, pid=424242)

